src/Comments/Comments.jsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from "react";
+import { fetchCommentById } from "../api";
+import { dateFormatter } from "../utils";
+import "./comments.css";
+
+const Comments = ({ article_id }) => {
+  const [comments, setComments] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    fetchCommentById(article_id).then((fetchedComments) => {
+      setComments(fetchedComments);
+      setIsLoading(false);
+    });
+  });
+
+  if (isLoading) {
+    return <p className="isLoading">Loading comments...</p>;
+  }
+
+  if (comments.length === 0) {
```

Comment:
This is sound logic unfortunately it seems you don't get an empty array back from your API in the instance of no comments - comments are undefined. You can either update your API (through TDD of course) to fix this or use some conditional logic on the front end to set the comments to be the required empty array in the useEffect based on what _does_ come back from your API. You will need to alter your fetchCommentById function to return not just the comments property to achieve this.